client/src/common/client.ts

```diff
@@ -1814,6 +1816,15 @@ export abstract class BaseLanguageClient implements FeatureClient<Middleware, La
 	}
 
 	private async handleRegistrationRequest(params: RegistrationParams): Promise<void> {
+		const middleware = this.clientOptions.middleware?.handleRegisterCapability;
+		if(middleware) {
+			return middleware(params, nextParams => this.tryToRegisterCapability(nextParams));
+		} else {
+			return this.tryToRegisterCapability(params);
+		}
+	}
+
+	private async tryToRegisterCapability(params: RegistrationParams): Promise<void> {
```

Comment:
I would not call this `try*` since it is not a try in the classical sense. I would suggest `do*`

Reply:
Fixed!